### PR TITLE
[browser] JSObject thread affinity assertions

### PR DESCRIPTION
**Pull request by pavelsavara**
- add asserts for JSObject and JSException that they could be used only on original thread where they are bound to JS.
- add asserts that legacy interop only works on main thread - ID 1

### CODE REVIEW

**Comment by pavelsavara** [comment]
CI failures are unrelated

**Review by lambdageek — APPROVED** [review]

File: src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSObject.References.cs [hunk 1]
```diff
@@ -56,6 +64,30 @@ internal void ReleaseInFlight()
...
+                if (jsObject.OwnerThreadId != Thread.CurrentThread.ManagedThreadId)
+                {
+                    throw new InvalidOperationException("The JavaScript object can be used only on the thread where it was created.");
+                }
+            }
+            if (value is JSException jsException)
+            {
+                if(jsException.jsException!=null && jsException.jsException.OwnerThreadId != Thread.CurrentThread.ManagedThreadId)
+                {
+                    throw new InvalidOperationException("The JavaScript object can be used only on the thread where it was created.");
```
Comment (ilonatommy) [inline src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSObject.References.cs:85]: We don't want to surface the exception that the JsException contains? Why the message is 100% the same as for JsObject?
```suggestion
                    throw new InvalidOperationException("The JavaScript exception can be used only on the thread where it was created.");
```

File: src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Legacy/LegacyHostImplementation.cs [hunk 2]
```diff
@@ -220,5 +221,15 @@ public enum MappedType
             Function = 4,
             Uint8Array = 11,
         }
+
+#if FEATURE_WASM_THREADS
+        public static void ThrowIfLegacyWorkerThread()
+        {
+            if (Thread.CurrentThread.ManagedThreadId != 1)
+            {
+                throw new PlatformNotSupportedException("Legacy interop in not supported with WebAssembly threads."+Thread.CurrentThread.ManagedThreadId);
```
Comment (ilonatommy) [inline src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Legacy/LegacyHostImplementation.cs:230]: Small typo:
```suggestion
                throw new PlatformNotSupportedException("Legacy interop is not supported with WebAssembly threads."+Thread.CurrentThread.ManagedThreadId);
```

File: src/mono/wasm/runtime/net6-legacy/method-binding.ts [hunk 3]
```diff
@@ -382,6 +383,9 @@ export function _decide_if_result_is_marshaled(converter: Converter, argc: numbe
 
 
 export function mono_bind_method(method: MonoMethod, args_marshal: string/*ArgsMarshalString*/, has_this_arg: boolean, friendly_name?: string): Function {
+    if (MonoWasmThreads && ENVIRONMENT_IS_PTHREAD) {
+        throw new Error("Legacy interop in not supported with WebAssembly threads.");
```
Comment (ilonatommy) [inline src/mono/wasm/runtime/net6-legacy/method-binding.ts:387]:
```suggestion
        throw new Error("Legacy interop is not supported with WebAssembly threads.");
```

File: src/mono/wasm/runtime/net6-legacy/method-calls.ts [hunk 4]
```diff
@@ -90,6 +91,9 @@ export function mono_call_assembly_entry_point(assembly: string, args?: any[], s
 }
 
 export function mono_wasm_invoke_js_with_args_ref(js_handle: JSHandle, method_name: MonoStringRef, args: MonoObjectRef, is_exception: Int32Ptr, result_address: MonoObjectRef): any {
+    if (MonoWasmThreads && ENVIRONMENT_IS_PTHREAD) {
+        throw new Error("Legacy interop in not supported with WebAssembly threads.");
```
Comment (ilonatommy) [inline src/mono/wasm/runtime/net6-legacy/method-calls.ts:95]:
```suggestion
        throw new Error("Legacy interop is not supported with WebAssembly threads.");
```

File: src/mono/wasm/runtime/net6-legacy/method-calls.ts [hunk 5]
```diff
@@ -282,6 +286,9 @@ export function mono_wasm_get_global_object_ref(global_name: MonoStringRef, is_e
 // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
 export function mono_wasm_invoke_js_blazor(exceptionMessage: Int32Ptr, callInfo: any, arg0: any, arg1: any, arg2: any): void | number {
     try {
+        if (MonoWasmThreads) {
+            throw new Error("Legacy interop in not supported with WebAssembly threads.");
```
Comment (ilonatommy) [inline src/mono/wasm/runtime/net6-legacy/method-calls.ts:290]:
```suggestion
            throw new Error("Legacy interop is not supported with WebAssembly threads.");
```

**Review by ilonatommy — APPROVED** [review]
Looks good, only small copy-paste errors.

**Pull request merged**